sdks/java/io/mqtt/src/test/java/org/apache/beam/sdk/io/mqtt/MqttIOTest.java

```diff
@@ -193,6 +195,22 @@ public void run() {
     publisherThread.join();
   }
 
+  /**
+   * Test for BEAM-3282: this test should not timeout, and start/advance should return false.
+   */
+  @Test(timeout = 3 * 1000)
```

Comment:
That's an overly tight timeout and it might fail flakily. I'd recommend to set this one to like 5 seconds.